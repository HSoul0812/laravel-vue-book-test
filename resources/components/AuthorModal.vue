<template>
  <b-modal v-model="getShow" @ok="addAuthor">
    <h4 class="text-center">Input Author Info</h4>
    <form ref="form" @submit.stop.prevent class="px-5">
      <div v-for="(item, index) in headers" :key="item.text">
        <div class="d-flex align-items-center justify-content-between my-2">
          <div class="w-50">{{ item.text }} : </div>
          <b-form-input class="w-50" v-model="currentItem[`${item.value}`]" className='w-50 mx-auto'
            :disabled="index > 2" data-test="authorTest" required></b-form-input>
        </div>
      </div>
    </form>
  </b-modal>
</template>

<script>
export default {
  name: 'AuthorModal',
  props: {
    show: [Boolean],
  },
  data: function () {
    return {
      currentItem: {},
      headers: [
        { text: 'Author Name', value: "authorName" },
        { text: 'Genre', value: "authorGenre" },
        { text: 'Birthday', value: "authorBirth" },
      ],
    }
  },

  methods: {
    async addAuthor() {
      this.$emit('addItem', this.currentItem);
    },

  },
  computed: {
    getShow: {
      get() {
        return this.show
      },
      set(value) {
        this.$emit('updateShow', value)
      }
    },
  }
}
</script>

