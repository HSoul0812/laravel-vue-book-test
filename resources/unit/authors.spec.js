import { mount } from '@vue/test-utils'
import AuthorModal from '../components/AuthorModal'

describe('Author Modal Test', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(AuthorModal)
  })

  test('renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
