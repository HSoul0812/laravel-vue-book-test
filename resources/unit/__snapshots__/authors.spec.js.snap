// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Modal Test renders properly 1`] = `
"<b-modal modelvalue=\\"false\\">
  <h4 class=\\"text-center\\">Input Author Info</h4>
  <form class=\\"px-5\\">
    <div>
      <div class=\\"d-flex align-items-center justify-content-between my-2\\">
        <div class=\\"w-50\\">Author Name : </div>
        <b-form-input class=\\"w-50 mx-auto\\" disabled=\\"false\\" data-test=\\"authorTest\\" required=\\"\\"></b-form-input>
      </div>
    </div>
    <div>
      <div class=\\"d-flex align-items-center justify-content-between my-2\\">
        <div class=\\"w-50\\">Genre : </div>
        <b-form-input class=\\"w-50 mx-auto\\" disabled=\\"false\\" data-test=\\"authorTest\\" required=\\"\\"></b-form-input>
      </div>
    </div>
    <div>
      <div class=\\"d-flex align-items-center justify-content-between my-2\\">
        <div class=\\"w-50\\">Birthday : </div>
        <b-form-input class=\\"w-50 mx-auto\\" disabled=\\"false\\" data-test=\\"authorTest\\" required=\\"\\"></b-form-input>
      </div>
    </div>
  </form>
</b-modal>"
`;
